src/main/java/com/greedobank/cards/mapper/CardMapper.java

```diff
@@ -0,0 +1,51 @@
+package com.greedobank.cards.mapper;
+
+import com.greedobank.cards.dao.CardTemplateDAO;
+import com.greedobank.cards.dao.CustomerDAO;
+import com.greedobank.cards.dto.CardCreationDTO;
+import com.greedobank.cards.dto.CardDTO;
+import com.greedobank.cards.exception.NotFoundException;
+import com.greedobank.cards.model.Card;
+import com.greedobank.cards.model.CardTemplate;
+import com.greedobank.cards.model.Customer;
+import com.greedobank.cards.utils.CurrencyType;
+import com.greedobank.cards.utils.ResponseMessages;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.time.format.DateTimeFormatter;
+
+@Component
+@RequiredArgsConstructor
+public class CardMapper {
+    private final CardTemplateDAO cardTemplateDAO;
+    private final CustomerDAO customerDAO;
+
+    public Card toCard(CardCreationDTO cardCreationDTO) {
```

Comment:
why not using mapstruct?

Reply:
I agree that it good solution, but It wasn't in my tasks